x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_control_panels_url_state.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import * as rt from 'io-ts';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { fold } from 'fp-ts/lib/Either';
+import { constant, identity } from 'fp-ts/lib/function';
+import { ControlGroupInput } from '@kbn/controls-plugin/common';
+import { useUrlState } from '../../../../utils/use_url_state';
+
+export const getDefaultPanels = (dataViewId: string): ControlGroupInput['panels'] =>
+  ({
+    osPanel: {
+      order: 0,
+      width: 'medium',
+      grow: false,
+      type: 'optionsListControl',
+      explicitInput: {
+        id: 'osPanel',
+        dataViewId,
+        fieldName: 'host.os.name',
+        title: 'Operating System',
+      },
+    },
+    cloudProviderPanel: {
+      order: 1,
+      width: 'medium',
+      grow: false,
+      type: 'optionsListControl',
+      explicitInput: {
+        id: 'cloudProviderPanel',
+        dataViewId,
+        fieldName: 'cloud.provider',
+        title: 'Cloud Provider',
+      },
+    },
+  } as unknown as ControlGroupInput['panels']);
+const HOST_FILTERS_URL_STATE_KEY = 'controlPanels';
+
+export const useControlPanels = (dataViewId: string) => {
+  const [urlState, setUrlState] = useUrlState<ControlPanels>({
+    defaultState: getDefaultPanels(dataViewId),
+    decodeUrlState,
+    encodeUrlState,
+    urlStateKey: HOST_FILTERS_URL_STATE_KEY,
+  });
+
+  const setControlPanels = useCallback(
+    (panels: ControlPanels) => {
+      setUrlState(panels);
+    },
+    [setUrlState]
+  );
```

Comment:
**suggestion**: in my understanding (correct me if wrong, please), the useUrlState already returns a memorized handler `setUrlState`. This declaration is unnecessary since the function only forwards the argument to the updated handler. 

You could rename it directly with a new variable and keep the same function reference:
```ts
const setControlPanels = setUrlState
```

Reply:
Good point! I don't need the `useCallback` anymore 🙂  I set it directly in the `return` same as the `urlState` so it will be even simpler: 
```
return {
    controlPanel: urlState,
    setControlPanels: setUrlState,
}
```